editor/src/clj/editor/fuzzy_text.clj

```diff
@@ -226,69 +242,95 @@
       match-a
       match-b)))
 
+(defn- match-impl
+  [^String upper-pattern ^String lower-pattern ^String string ^long from-index]
+  (transduce
+    (map (fn [matching-indices]
+           (pair (score string from-index matching-indices)
+                 matching-indices)))
+    (fn reduce-fn
+      ([best-match] best-match)
+      ([match-a match-b] (best-match match-a match-b)))
+    nil
+    (matching-index-permutations upper-pattern lower-pattern string from-index)))
+
+(defn prepare-pattern
+  "Given a pattern string, returns a trimmed and case-agnostic prepared-pattern
+  for use with the match functions."
+  [^String pattern]
+  (let [trimmed-pattern (string/trim pattern)]
+    (pair (string/upper-case trimmed-pattern)
+          (string/lower-case trimmed-pattern))))
+
+(defn empty-prepared-pattern?
+  "Returns true if the supplied prepared-pattern is empty."
+  [prepared-pattern]
+  (let [^String upper-pattern (key prepared-pattern)]
+    (.isEmpty upper-pattern)))
+
 (defn match
-  "Performs a fuzzy text match against a string using the specified pattern.
+  "Performs a fuzzy text match against a string using the prepared-pattern.
   Returns a two-element vector of [score, matching-indices], or nil if the
-  pattern is empty or there is no match. The matching-indices vector will
-  contain the character indices in string that matched the pattern in sequential
-  order. A lower score represents a better match."
-  ([^String pattern ^String string]
-   (match pattern string 0))
-  ([^String pattern ^String string ^long from-index]
-   (transduce (map (fn [matching-indices]
-                     (pair (score string from-index matching-indices)
-                           matching-indices)))
-              (completing best-match)
-              nil
-              (matching-index-permutations pattern string from-index))))
+  pattern is empty or there is no match. The matching-indices bit-set will
+  contain the character indices in the string that matched the pattern in
+  sequential order. A lower score represents a better match."
+  [prepared-pattern ^String string]
+  (let [[upper-pattern lower-pattern] prepared-pattern]
+    (match-impl upper-pattern lower-pattern string 0)))
 
 (defn- apply-filename-bonus
   "Applies a bonus for a match on the filename part of a path."
-  [^String path ^long basename-start [^long score matched-indices]]
+  [^String path ^long basename-start [^long score matching-indices]]
```

Comment:
Renamed argument to `matching-indices`, which is the term we use for it everywhere else.